msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/nativeauth/SignInTest.kt

```diff
@@ -43,11 +91,57 @@ class SignInTest : NativeAuthPublicClientApplicationAbstractTest() {
         Assert.assertTrue((result as SignInError).isInvalidCredentials())
     }
 
+    /**
+     * Use email and OTP to get token and sign in (hero scenario 6, use case 2.2.1) - Test case 30
+     */
     @Test
-    fun testSignInSuccessSimple() = runTest {
-        val username = NativeAuthCredentialHelper.nativeAuthSignInUsername
-        val password = getSafePassword()
-        val result = application.signIn(username, password.toCharArray())
-        Assert.assertTrue(result is SignInResult.Complete)
+    fun testSuccessEmailOTP() {
+        var signInResult: SignInResult
+        var otp: String
+
+        retryOperation {
+            runBlocking {
+                val user = NativeAuthCredentialHelper.nativeAuthSignInUsername
```

Comment:
wait, how is this passing? The existing user is a password user, so they shouldn't be getting "CodeRequired". AFAIK in order to support email + OTP we'll need:
- a new user
- new env vars and BuildConfig support

Reply:
Agreed. Should I include those in this PR? I personally think a new one is required specific for the config. Another idea is that we are adding a lots of parameters into the command script, maybe be we can aggregate them into the <config>

Reply:
Up to you how you want to submit them PR-wise. We won't need a different tenant config to use a different auth-type user (sign in auth is at a user-level, not application level. only sign up is on application level), but we will indeed need a different config for email-OTP sign up flows. 

I agree re. the parameters; we may need a better solution for that. let's discuss with the iOS team.